src/silx/gui/dialog/ColormapDialog.py

```diff
@@ -1195,10 +1326,21 @@ def setItem(self, item):
         the data range or the histogram of the data using :meth:`setDataRange`
         and :meth:`setHistogram`
         """
-        # While event from items are not supported, we can't ignore dup items
-        # old = self._getItem()
-        # if old is item:
-        #     return
+        old = self._getItem()
+        if old is item:
+            # While event from items are not supported, we can't ignore dup items
+            if item is not None:
+                array = item.getColormappedData(copy=False)
+            else:
+                array = None
+            colormapped = self._colormapped
+            if colormapped is not None:
+                oldArray = colormapped()
+            else:
+                oldArray = None
+            if oldArray is array:
+                return
```

Comment:
It's usually not a good idea to rely on `is` to ensure the array content didn't changed... but in this case, looking at `getColormappedData` and `_setColormappedData` it should work since a new `numpy.ndarray` object is created each time, so it should work but it can be fragile.